client/src/app/core/services/data-send.service.ts

```diff
@@ -16,38 +14,47 @@ export class DataSendService {
     /**
      * Construct a DataSendService
      *
-     * @param httpService The HTTP Client
+     * @param httpService The HTTP Service
      */
     public constructor(private httpService: HttpService) {}
 
     /**
-     * Sends a post request with the model to the server.
-     * Usually for new Models
+     * Sends a post request with the model to the server to create it.
+     *
+     * @param model The model to create.
      */
-    public createModel(model: BaseModel): Observable<BaseModel> {
+    public async createModel<T>(model: BaseModel): Promise<T> {
```

Comment:
What exactly is the advantage of a Promise in contradictiopn of an Observable?

I get the concept, but this functions run in the background anyway and are most likely displayed by Observables again. So why put the whole DOM to a halt for the await of a promise?

Wouldn't this eventually hurt in the offline-mode?

Reply:
This should be  `<void>` as well `id`